cylc/flow/cfgspec/suite.py

```diff
@@ -1326,6 +1324,16 @@ def upg(cfg, descr):
     # and [job] moved up 1 level for example) which should be upgraded here.
     u.upgrade()
 
+    # Move an option from config to command line:
+    if ('cylc' in cfg and
+        'abort if any task fails' in cfg['cylc']
+    ):
+        LOG.warning(
+            "The config item ['cylc']['abort if any task fails'] is now the "
+            "command line option '--abort-if-any-task-fails', and will be "
+            "removed at Cylc 9."
+        )
+
```

Comment:
The config proposal says to obsolete this setting rather than deprecate it.

Reply:
I've made this unnecessarily hard for myself...

Reply:
Now just need to remove the old syntax from the tests and execute `cylc (run|restart)` with the new CLI opt where necessary.